src/app.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useReducer } from 'react';
+import { st, classes } from './app.st.css';
+import { TodoList } from './components/TodoList';
+import { AddTodo } from './components/AddTodo';
+import type { Todo } from './models/todo.model';
+import { Header } from './header';
+
+export interface AppProps {
+    className?: string;
+}
+
+export const ACTIONS = {
+    ADD_TODO: 'addTodo',
+    DELETE_TODO: 'deleteTodo',
+};
+
+export type Actions =
+    | { type: 'addTodo'; payload: { id: string; text: string } }
+    | { type: 'deleteTodo'; payload: { id: string } };
+
+export interface Action {
+    type: string;
+    payload?: Todo;
```

Comment:
I would make the payload an object that hold `todo` as a property so later on you can change add more properties without breaking changes